From 8facae7017b1e1ca6aac48f1712b131659334cd3 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 24 Jun 2025 18:01:32 +0800
Subject: [PATCH] ksavefile.cpp: undefine HAVE_FDATASYNC on Apple, it does
 not work

kdecore/io/ksavefile.cpp:220:23: error: no member named 'fdatasync' in the global namespace

---
 kdecore/io/ksavefile.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/kdecore/io/ksavefile.cpp b/kdecore/io/ksavefile.cpp
index a899fa00..81b26607 100644
--- a/kdecore/io/ksavefile.cpp
+++ b/kdecore/io/ksavefile.cpp
@@ -40,6 +40,10 @@
 #include <stdlib.h>
 #include <errno.h>
 
+#ifdef __APPLE__
+#undef HAVE_FDATASYNC
+#endif
+
 class KSaveFile::Private
 {
 public:
